### PR TITLE
[api][webui] Add nil as default for Backend#post

### PR DESCRIPTION
**Pull request by Ana06**
Add `nil` as default for the second argument of `Suse::Backend#post` as it is used a lot of times with `nil` or `''`.

### CODE REVIEW

File: src/api/app/models/package.rb [hunk 1]
```diff
@@ -1323,7 +1323,7 @@ def backend_build_command(command, params)
       # Note: This list needs to keep in sync with the backend code
       permitted_params = params.permit(:repository, :arch, :package, :code, :wipe)
 
-      Suse::Backend.post("/build/#{URI.escape(project.name)}?cmd=#{command}&#{permitted_params.to_h.to_query}", '')
+      Suse::Backend.post("/build/#{URI.escape(project.name)}?cmd=#{command}&#{permitted_params.to_h.to_query}")
```
Comment (ChrisBr) [inline src/api/app/models/package.rb]: Is ``nil`` and ``''`` the same in this context?

File: src/api/app/models/package.rb [hunk 2]
```diff
@@ -471,7 +471,7 @@ def update_issue_list
 
   def parse_issues_xml(query, force_state = nil)
     begin
-      answer = Suse::Backend.post(self.source_path(nil, query), nil)
+      answer = Suse::Backend.post self.source_path(nil, query)
```
Comment (ChrisBr) [inline src/api/app/models/package.rb]: Why did you remove the parentheses?

File: src/api/app/controllers/wizard_controller.rb [hunk 3]
```diff
@@ -56,7 +56,7 @@ def package_wizard
     xml = create_service_file
 
     Suse::Backend.put("/source/#{params[:project]}/#{params[:package]}/_service?rev=upload&user=#{User.current.login}", xml)
-    Suse::Backend.post("/source/#{params[:project]}/#{params[:package]}?cmd=commit&rev=upload&user=#{User.current.login}", '')
+    Suse::Backend.post("/source/#{params[:project]}/#{params[:package]}?cmd=commit&rev=upload&user=#{User.current.login}")
```
Comment (ChrisBr) [inline src/api/app/controllers/wizard_controller.rb]: Is ``nil`` and ``''`` the same in this context?

File: src/api/app/models/package.rb (hunk 2 above)
Comment (Ana06) [inline src/api/app/models/package.rb]: A lot of parenthesis, I thought it was more readable without them, but I can add them again if you want. :no_mouth:

File: src/api/app/models/package.rb (hunk 1 above)
Comment (Ana06) [inline src/api/app/models/package.rb]: I think so, because this is assign to the backend request body: 
https://github.com/Ana06/open-build-service/blob/0a821509502fb794a5d125f4f8ec9e2af386f53a/src/api/lib/opensuse/backend.rb#L76

And the tests are not failing.

File: src/api/app/controllers/wizard_controller.rb (hunk 3 above)
Comment (adrianschroeter) [inline src/api/app/controllers/wizard_controller.rb]: it is not the same, but nil is the more correct one here actually...

Comment (Ana06) [inline src/api/app/controllers/wizard_controller.rb]: why is not the same?

File: src/api/app/models/package.rb (hunk 1 above)
Comment (adrianschroeter) [inline src/api/app/models/package.rb]: _it is not the same, but nil is the more correct one here ;)

Comment (Ana06) [inline src/api/app/models/package.rb]: @adrianschroeter why is not the same?

File: src/api/app/controllers/wizard_controller.rb (hunk 3 above)
Comment (adrianschroeter) [inline src/api/app/controllers/wizard_controller.rb]: On Donnerstag, 22. September 2016, 05:55:30 CEST wrote Ana Mar�a Mart�nez G�mez:
> > @@ -56,7 +56,7 @@ def package_wizard
> >      xml = create_service_file
> >  
> >      Suse::Backend.put("/source/#{params[:project]}/#{params[:package]}/_service?rev=upload&user=#{User.current.login}", xml)
> > -    Suse::Backend.post("/source/#{params[:project]}/#{params[:package]}?cmd=commit&rev=upload&user=#{User.current.login}", '')
> > +    Suse::Backend.post("/source/#{params[:project]}/#{params[:package]}?cmd=commit&rev=upload&user=#{User.current.login}")
> 
> @adrianschroeter why is not the same?

'' will create a body, even though it will be empty.
It also has some effects on how we do logging backend 
requests. Have a look in write_backend_log() in 

 src/api/lib/opensuse/backend.rb

if you are interessted in that.

-- 

Adrian Schroeter
email: adrian@suse.de

SUSE Linux GmbH, GF: Felix Imend�rffer, Jane Smithard, Graham Norton, HRB 21284 (AG N�rnberg) 
 
Maxfeldstra�e 5                         
90409 N�rnberg 
Germany

Comment (Ana06) [inline src/api/app/controllers/wizard_controller.rb]: @adrianschroeter thank you.

File: src/api/app/models/package.rb (hunk 1 above)
Comment (bgeuken) [inline src/api/app/models/package.rb]: Because of https://github.com/openSUSE/open-build-service/blob/master/src/api/lib/opensuse/backend.rb#L72

Though if that matters here... I dunno

Comment (Ana06) [inline src/api/app/models/package.rb]: It is the same for `nil` and `''`:
```
irb(main):001:0> nil.respond_to?('read')
=> false
irb(main):002:0> ''.respond_to?('read')
=> false
```

Or do you mean other thing?

File: src/api/app/models/package.rb (hunk 2 above)
Comment (mdeniz) [inline src/api/app/models/package.rb]: In this case I agree with Chris... keep the parenthesis :+1:

Comment (Ana06) [inline src/api/app/models/package.rb]: :+1:

File: src/api/app/models/package.rb (hunk 1 above)
Comment (bgeuken) [inline src/api/app/models/package.rb]: Oh, right. I've seen Adrian already replied on this one. So let's merge it.
